edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/logs/ILogsProcessor.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Logs
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+
+    public interface ILogsProcessor
+    {
+        Task<IEnumerable<ModuleLogMessage>> GetMessages(Stream stream, string moduleId);
```

Comment:
Return Task<IEnumerable> looks weird to me; check this out https://what.thedailywtf.com/topic/22986/async-await-with-ienumerable-in-c.  Do we really need Task<IEnumerable> ? or just use Task<IList> or Task<ICollection>?

Reply:
So your concern is it loads the entire list? That is currently by design for this method - it materializes the enitre list before returning. 
I am adding another one, where it will process the bytes without returning them.

Reply:
If this is actually returning a materialized list by design maybe we should make indicate the eagerness by using `ICollection` or `IList`?

Reply:
Sure, makes sense, I can do that.